---
title: "Creatinine & Cystatin C & CKD as possible consequences of FMD"
format: gfm
---


```{r}
library(magrittr)
```


```{r}
fmd_file <- here::here("analysis", "data", "fmd", "GCST90026612_buildGRCh37.tsv")
fmd <- TwoSampleMR::read_exposure_data(filename = fmd_file,
                                        sep = "\t",
                                        snp_col = "SNP",
                                        beta_col = "BETA",
                                        effect_allele_col = "EA",
                                        other_allele_col = "OA",
                                        eaf_col = "EAF",
                                        se_col = "SE",
                                        pval_col = "p_value",
                                        chr_col = "chromosome",
                                        pos_col = "base_pair_location",
                                        ncase_col = "N_cases",
                                        ncontrol_col = "N_ctrls"
                                        )
pvalue_threshold <- 1e-6
```



```{r, results = "asis"}
files <- list.files(here::here("analysis", "data", "ukb_for_munge_sumstats"), pattern = "female", full.names = TRUE)
file <- files[1]
pre_dat <- vroom::vroom(file, col_types = "ciiccdddi")
dat <- fmd %>% 
    dplyr::filter(pval.exposure < pvalue_threshold) %>%
    TwoSampleMR::clump_data() %>%
    dplyr::mutate(chr_pos = paste0(chr.exposure, ":", pos.exposure)) %>%
    dplyr::left_join(pre_dat, by = c("chr_pos" = "snp")) 
dat2 <- dat %>%
    tibble::as_tibble() %>%
    dplyr::select(- c(chr_pos, chr, pos)) %>%
    dplyr::rename(effect_allele.outcome = A1, 
                    other_allele.outcome = A2, 
                    beta.outcome =  beta, 
                    se.outcome = se, 
                    pval.outcome = pvalue, 
                    n.outcome = n) %>%
    dplyr::relocate(other_allele.exposure, 
                    other_allele.outcome, 
                    effect_allele.exposure, 
                    effect_allele.outcome) 
# now use dat2 with TwoSampleMR functions to get MR results





# see neale lab documentation: https://docs.google.com/spreadsheets/d/1kvPoupSzsSFBNSztMzl04xMoSC3Kcx3CrjVf4yBmESU/edit#gid=227859291
# GRCh37 is used here!
#Neale lab denotes by beta the alt allele effect, ie, the increase in phenotype per increase in one alt allele.


```



```{r}
sessioninfo::session_info()
# git commit info
gr <- git2r::repository(here::here()) %>%
    git2r::commits()
gr[[1]] 
```